app/scripts/runLockdown.js

```diff
@@ -0,0 +1,7 @@
+// Freezes all intrinsics
+// eslint-disable-next-line no-undef,import/unambiguous
+lockdown({
+  errorTaming: 'unsafe',
```

Comment:
So by setting these three to "unsafe", are they completely unaffected by the lockdown?

I tried looking this up in the SES docs, but I couldn't find where this was documented.

I was going to ask about whether Errors still had stack traces now, as I had read that SES disabled stack traces for security reasons, and had some alternate way to get them. But I'm guessing that this leaves them exactly as-is for now, which is good.

Reply:
this part of SES has changed recently and so i would not consider it stable, but for now:
- the `errorTaming: 'unsafe'` allows you to retain error stacks. error stacks are normally hidden so that code cant observe the context in which its called. ses has a feature where the error stack is hidden but can be recovered for eg logging to console. this feature is not exposed yet so we cant use it to expose to sentry.
- all other `*Taming: 'unsafe'` allow potentially powerful/dangerous but expected things like non-determinism via `Math.random` and `new Date()`
- we leave security on for `RegExp.lastMatch` as it is not used much in practice

Reply:
more fine grained control over access to features like Math.random will come with LavaMoat 🌋

Reply:
That is correct. Setting `errorTaming` to `unsafe` reveals stack traces. Math is for `math.random` & Date is for `Date.now`. They aren't unaffected by lockdown, but instead accessible via the global namespace. SES disables access by default for security concerns.